gdal/swig/python/gdal-utils/osgeo_utils/auxiliary/raster_creation.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# ******************************************************************************
+#
+#  Project:  GDAL utils.auxiliary
+#  Purpose:  raster creation utility functions
+#  Author:   Idan Miara <idan@miara.com>
+#
+# ******************************************************************************
+#  Copyright (c) 2021, Idan Miara <idan@miara.com>
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a
+#  copy of this software and associated documentation files (the "Software"),
+#  to deal in the Software without restriction, including without limitation
+#  the rights to use, copy, modify, merge, publish, distribute, sublicense,
+#  and/or sell copies of the Software, and to permit persons to whom the
+#  Software is furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included
+#  in all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+#  OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+#  THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+#  DEALINGS IN THE SOFTWARE.
+# ******************************************************************************
+
+import tempfile
+from numbers import Real
+from typing import Sequence, Optional, Union
+
+from osgeo import gdal, osr
+from osgeo_utils.auxiliary.base import PathLike, MaybeSequence
+from osgeo_utils.auxiliary.util import NO_YES, get_big_tiff, get_tiled, get_data_type, DataTypeOrStr, CreationOptions
+
+
+def create_flat_raster(filename: Optional[PathLike],
+                       driver: str = 'GTiff', dt: DataTypeOrStr = gdal.GDT_Byte,
+                       size: MaybeSequence[int] = 128, band_count: int = 1, creation_options: CreationOptions = None,
+                       fill_value: Optional[Real] = None, nodata_value: Optional[Real] = None,
+                       origin: Optional[Sequence[int]] = None, pixel_size: MaybeSequence[int] = 10,
+                       epsg: Optional[int] = 32636,
+                       overview_alg: Optional[str] = 'NEAR', overview_list: Optional[Sequence[int]] = None):
+    if filename is None:
+        filename = tempfile.mktemp()
+    if not isinstance(pixel_size, Sequence):
+        pixel_size = (pixel_size, -pixel_size)
```

Comment:
pixel_size[1] is negated afterwards, so it is not needed here

Reply:
thanks, I removed the negative sign on SetGeoTransform